core/tee/tee_svc_storage.c

```diff
@@ -834,22 +834,29 @@ TEE_Result tee_svc_storage_alloc_enum(uint32_t *obj_enum)
 	struct tee_ta_session *sess;
 	TEE_Result res;
 
-	if (obj_enum == NULL)
-		return TEE_ERROR_BAD_PARAMETERS;
+	if (!obj_enum) {
+		res = TEE_ERROR_BAD_PARAMETERS;
+		goto exit;
```

Comment:
And at the same time become inconsistent with other functions... :-)

The `goto exit` technique is very good when a function need to do some cleanup before returning, but if there's not cleanup to be done it doesn't bring any value. Rather the opposite actually. 

We probably have some functions where we do `goto exit` without any cleanup there, but it's usually due to that the function has changed since it was originally written.